protocols/gossipsub/src/behaviour.rs

```diff
@@ -3177,6 +3267,24 @@ where
                             peers,
                             backoff,
                         }) => prune_msgs.push((topic_hash, peers, backoff)),
+                        ControlAction::IDontWant(IDontWant { message_ids }) => {
+                            let Some(peer) = self.connected_peers.get_mut(&propagation_source)
+                            else {
+                                tracing::error!(peer = %propagation_source,
+                                    "Could not handle IDONTWANT, peer doesn't exist in connected peer list");
+                                continue;
+                            };
+                            if let Some(metrics) = self.metrics.as_mut() {
+                                metrics.register_idontwant(message_ids.len());
+                            }
+                            for message_id in message_ids {
+                                peer.dont_send.insert(message_id, Instant::now());
```

Comment:
What happens if the message_id is already in the map? Is it just updated, or also move to back?
I'm asking because we want to pop the least recently seen message id below if we exceed capacity, right? If so, should we first remove here the message id, to ensure that it is reinserted at the back?

Reply:
Actually, I think we must do this, otherwise the flushing logic won't work correctly.

Reply:
A [comment](https://github.com/kyren/hashlink/blob/11088cb368f93e1f61fc42e696b4ab89158ae70d/src/linked_hash_map.rs#L319)  from the hashlink crate states that this behaviour is inherent to the data structure itself, if I understand correctly

Reply:
Ah yes you're completely right. I was looking at the wrong crate `linked-hash-map`.